src/dynamixel/protocols/protocol1.hpp

```diff
@@ -113,14 +113,14 @@ namespace dynamixel {
             static void unpack_data(const std::vector<uint8_t>& packet, uint8_t& res)
             {
                 if (packet.size() == 0)
-                    throw errors::UnpackError(1);
+                    throw errors::UnpackError(1, packet.size(), 0);
```

Comment:
Why the expected size here is zero?

Reply:
It looks like a mistake. Do you agree with "1" as the right value ?

Also, I wonder now why the condition is not `packet.size() != 1`

Reply:
Yeap. The condition should be `packet.size() != 1` and "1" seems like the right value..